mllib/src/test/scala/org/apache/spark/ml/util/DefaultReadWriteTest.scala

```diff
@@ -113,7 +115,14 @@ trait DefaultReadWriteTest extends TempDirectory { self: Suite =>
     val estimator2 = testDefaultReadWrite(estimator)
     testEstimatorParams.foreach { case (p, v) =>
       val param = estimator.getParam(p)
-      assert(estimator.get(param).get === estimator2.get(param).get)
+      if (param.name == "initialModel") {
+        // Estimator's `initialModel` has same type as the model produced by this estimator.
```

Comment:
This is an assumption that ```initialModel``` must be the same type of the model that the estimator produced. We should enforce this assumption by making the trait as ```HasInitialModel[T <: Model[T]]``` or add new argument ```checkInitialModelData``` for ```testEstimatorAndModelReadWrite```. I'm open to hear others' thoughts and update here after collecting feedback. Thanks.